scripts/build.sh

```diff
@@ -138,39 +165,47 @@ function usage()
 #
 function install_cli()
 {
-    local failed=false
-    local install_script="$TP_TOOLS_DIR/dotnet-install.sh"
-    local remote_path="https://raw.githubusercontent.com/dotnet/cli/master/scripts/obtain/dotnet-install.sh"
-
-    log "Installing dotnet cli..."
-    local start=$SECONDS
+    if [[ $TP_USE_REPO_API ]]; then
+        # Skip download of dotnet toolset if REPO API is enabled
+        local failed=false
+        local install_script="$TP_TOOLS_DIR/dotnet-install.sh"
+        local remote_path="https://raw.githubusercontent.com/dotnet/cli/master/scripts/obtain/dotnet-install.sh"
+
+        log "Installing dotnet cli..."
+        local start=$SECONDS
+
+        # Install the latest version of dotnet-cli
+        curl --retry 10 -sSL --create-dirs -o $install_script $remote_path || failed=true
+        if [ "$failed" = true ]; then
```

Comment:
Why can't use `$?`

Reply:
https://unix.stackexchange.com/questions/325705/why-is-pattern-command-true-useful